web/src/components/tasks/NaicsCodeTask.test.tsx

```diff
@@ -417,14 +417,53 @@ describe("<NaicsCodeTask />", () => {
       expect(screen.getByText(`Register & ${Config.determineNaicsCode.saveButtonText}`)).toBeInTheDocument();
     });
 
-    it("opens Needs Account modal on save button click", async () => {
+    it("opens Needs Account modal on NAICS Code radio button click", async () => {
+      initialBusiness = generateBusiness({
+        profileData: generateProfileData({ naicsCode: "", industryId: "acupuncture" }),
+      });
       renderPage();
-      fireEvent.change(screen.getByLabelText("Save NAICS Code"), {
-        target: { value: validNaicsCode },
+
+      fireEvent.click(screen.getByTestId("naics-radio-621399"));
+
+      expect(setShowNeedsAccountModal).toHaveBeenCalledWith(true);
+      await waitFor(() => {
+        expect(screen.getByTestId("naics-radio-621399")).not.toBeChecked();
       });
-      fireEvent.click(screen.getByText(`Register & ${Config.determineNaicsCode.saveButtonText}`));
+    });
+
+    it("opens Needs Account modal on NAICS Code input radio button click", async () => {
+      initialBusiness = generateBusiness({
+        profileData: generateProfileData({ naicsCode: "", industryId: "acupuncture" }),
+      });
+      renderPage();
+
+      fireEvent.click(screen.getByTestId("naics-radio-input"));
+
+      expect(setShowNeedsAccountModal).toHaveBeenCalledWith(true);
       await waitFor(() => {
-        return expect(setShowNeedsAccountModal).toHaveBeenCalledWith(true);
+        expect(screen.getByTestId("naics-radio-input")).not.toBeChecked();
+      });
+    });
+
+    it("keeps task progress as NOT_STARTED when radio button is clicked", () => {
+      initialBusiness = generateBusiness({
+        profileData: generateProfileData({ naicsCode: "", industryId: "acupuncture" }),
+      });
+      renderPage();
+      fireEvent.click(screen.getByTestId("naics-radio-621399"));
+      expect(userDataWasNotUpdated()).toBe(true);
+    });
+
+    describe("guest mode - generic industry", () => {
+      it("opens modal when a user types in NAICS code input field", async () => {
+        initialBusiness = generateBusiness({
+          profileData: generateProfileData({ naicsCode: "", industryId: "generic" }),
+        });
+        renderPage();
+
+        fireEvent.change(screen.getByLabelText("Save NAICS Code"), { target: { value: "123456" } });
+
+        await expect(setShowNeedsAccountModal).toHaveBeenCalledWith(true);
```

Comment:
This is missing the expectation that nothing gets typed into the radio field. Please add the following assertion as well:

```
expect(screen.getByLabelText("Save NAICS Code")).toHaveValue('');
```

Also note that the `await` is unnecessary on the line above and should be removed.